### PR TITLE
RDK-43314:Hdmi EDID Allm support

### PR DESCRIPTION
**Pull request by aishwariya15**
Reason for change:Added the Wpeframework and the devicesettings Changes Test Procedure: NA
Risks: None
Signed-off-by: Aishwariya B aishwariya.bhaskar@sky.uk

### CODE REVIEW

**Review by apatel859 — CHANGES_REQUESTED** [review]
Update json documentgation. Also we need to check why API named as Edid2Allm

File: AVInput/AVInput.json [hunk 1]
```diff
@@ -175,6 +175,39 @@
...
+                },
+                "required": [
+                    "portid"
+                ]
+            },
+            "result": {
+                "type": "object",
+                "properties": {
+                    "allmSupport": {
+                        "summary": "The allm bit value in edid",
```
Comment (apatel859) [inline AVInput/AVInput.json:197]: could you use ALLM consistently

**Review by apatel859 — CHANGES_REQUESTED** [review]

File: AVInput/AVInput.json [hunk 2]
```diff
@@ -175,6 +175,39 @@
                 }
             ]
         },
+	"getEdid2AllmSupport":{
+            "deprecated" : true,
+            "referenceUrl" : "https://rdkcentral.github.io/rdkservices/#/api/AVInputPlugin?id=getEdid2AllmSupport",
+            "summary": "Returns the EDID Allm bit value.",
```
Comment (apatel859) [inline AVInput/AVInput.json:181]: ALLM

File: AVInput/AVInput.cpp [hunk 3]
```diff
@@ -45,6 +45,8 @@
 #define AVINPUT_METHOD_READ_SPD "getSPD"
 #define AVINPUT_METHOD_SET_EDID_VERSION "setEdidVersion"
 #define AVINPUT_METHOD_GET_EDID_VERSION "getEdidVersion"
+#define AVINPUT_METHOD_SET_EDID_ALLM_SUPPORT "setEdid2AllmSupport"
```
Comment (apatel859) [inline AVInput/AVInput.cpp:48]: where is the gunit test coverage for this two new method

File: Tests/mocks/devicesettings/HdmiInputMock.h [hunk 4]
```diff
@@ -17,6 +17,8 @@ class HdmiInputImplMock : public device::HdmiInputImpl {
     MOCK_METHOD(void, getHDMISPDInfo, (int iHdmiPort, std::vector<uint8_t> &data), (const, override));
     MOCK_METHOD(void, setEdidVersion, (int iHdmiPort, int iEdidVersion), (const, override));
     MOCK_METHOD(void, getEdidVersion, (int iHdmiPort, int *iEdidVersion), (const, override));
+    MOCK_METHOD(void, setEdid2AllmSupport, (int iHdmiPort, bool allmSupport), (const, override));
```
[the remark this answers is not in the document]
Comment (aishwariya15) [reply Tests/mocks/devicesettings/HdmiInputMock.h:20]: Hi, 
There is no mock header directly available for AVInputMock.h, Should we need to create it?

**Review by apatel859 — APPROVED** [review]

**Pull request merged**